File: Difficulty: Basic/Palindrome Array/palindrome-array.py

from typing import List
import copy

class Solution:
    def isPerfect(self, arr : List[int]) -> bool:
        arr1 = copy.deepcopy(arr)
        i = 0
        j = len(arr1)-1
        
        while (i<j):
            arr[i], arr[j] = arr[j], arr[i]
            i+=1
            j-=1
            
        if arr1 == arr:
            return True
            
        else:
            return False
        

File: Difficulty: Basic/Palindrome Array/palindrome-array.cpp
class Solution {
  public:
    bool isPerfect(vector<int> &arr) {
        int left = 0, right = arr.size() - 1;
        while (left < right) {
            if (arr[left] != arr[right]) {
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
};